src/hffs/fs.py

```diff
@@ -256,11 +245,15 @@ def ls(self, path, detail=True, **kwargs):
         return [o["name"] for o in out]
 
     def cp_file(self, path1, path2, **kwargs):
+        repo_type1, repo_id1, path_in_repo1 = self._resolve_repo_id(path1)
         path1 = self._strip_protocol(path1)
+        repo_type2, repo_id2, path_in_repo2 = self._resolve_repo_id(path2)
         path2 = self._strip_protocol(path2)
 
+        same_repo = repo_type1 == repo_type2 and repo_id1 == repo_id2
```

Comment:
I'd be careful if repo_type1 is None and repo_type2 is "model", it is still the same repo